src/main/java/com/amazon/opendistroforelasticsearch/sql/plugin/RestSqlAction.java

```diff
@@ -162,13 +163,14 @@ private static boolean isExplainRequest(final RestRequest request) {
 
     private static boolean isClientError(Exception e) {
         return e instanceof NullPointerException // NPE is hard to differentiate but more likely caused by bad query
-                || e instanceof SqlParseException
-                || e instanceof ParserException
-                || e instanceof SQLFeatureNotSupportedException
-                || e instanceof SQLFeatureDisabledException
-                || e instanceof IllegalArgumentException
-                || e instanceof IndexNotFoundException
-                || e instanceof VerificationException;
+            || e instanceof SqlParseException
```

Comment:
just an idea: if all the exceptions are derived from either `ClientException` or `ServerException`, then this function would just check if the exception is an instance/subclass of a `ClientException` and would be much faster.

Reply:
> just an idea: if all the exceptions are derived from either `ClientException` or `ServerException`, then this function would just check if the exception is an instance/subclass of a `ClientException` and would be much faster.

Good point. We should have better exception hierarchy. Will open an issue for tracking.

Reply:
Created issue: https://github.com/opendistro-for-elasticsearch/sql/issues/184